crates/core/tedge/src/cli/connect/bridge_config_c8y.rs

```diff
@@ -95,7 +95,9 @@ impl From<BridgeConfigC8yParams> for BridgeConfig {
             local_clean_session: false,
             notifications: true,
             notifications_local_only: true,
-            notification_topic: "tedge/health/mosquitto-c8y-bridge".into(),
+
+            // FIXME: doesn't account for custom topic root, use MQTT scheme API here
+            notification_topic: "te/device/main/service/mosquitto-c8y-bridge/status/health".into(),
```

Comment:
This constant is defined in `c8y_api` crate, which is currently not defined as dependency in `tedge` crate. IMO this constant should not be in `c8y_api` in the fist place, as I understand `c8y_api` should contain things relevant to the Cumulocity platform, like smartrest, while this topic is very much a local thin-edge matter.

So I will just redefine this constant in this module.

Reply:
Okay to let things as proposed here, the main argument is that it would be really bad to have a dependency from `tedge` to `c8y_api`.

However, longer term, I think it would be good to move all the c8y related code out of the `tedge` crate. With a trait that defines what tedge expects from a cloud plugin such as a bridge definition.

Reply:
> the main argument is that it would be really bad to have a dependency from `tedge` to `c8y_api`.

I wouldn't say it would be bad to have that dependency, only that "health topic of c8y" bridge is not something that belongs to `c8y_api` crate.

What I had in mind was for `tedge_api` and `c8y_api`, or any other `*_api` crate to only define models, data types, etc. and do not concern itself with effects like performing actions. `tedge_api` would be a single source of truth about local thin-edge API (e.g. "what messages are there", "what are entities and what can they do"), while `c8y_api` would express in Rust term and types what the Cumulocity cloud expects (e.g. "what are the different smartrest messages and how to create them", "what is inventory", "what are managed objects", etc.). And then all the "effects" would be performed by all the `_ext` crates, heavily utilizing relevant `_api` crates.

In this instance, I think `C8Y_BRIDGE_HEALTH_TOPIC` is just a health topic of `mosquitto-c8y-bridge` entity, so it's more of a `tedge_api/c8y_mapper_ext` mapper.


> However, longer term, I think it would be good to move all the c8y related code out of the `tedge` crate. With a trait that defines what tedge expects from a cloud plugin such as a bridge definition.

On that I heavily agree with. `tedge` crate itself is mainly just a CLI but it still handles setting up connections to different clouds, it's still a bit of a legacy code. It would be good to move all the bridge definitions to the relevant cloud mapper crates once we're no longer so pressed by features.